src/main/java/org/threadly/litesockets/tcp/TCPClient.java

```diff
@@ -30,119 +33,76 @@
-   * {@link SocketExecuterInterface#addClient(Client)} which will do the connect.
+   * yet made {@link #connect()} must be called which will do the actual connect.
    * 
-   * @param host hostname or ip address to connect too.
-   * @param port port on that host to try and connect too.
-   * @param timeout this is the max amount of time we will wait for a connection to be made.  The default is 10000 milliseconds (10 seconds).
+   * @param sei The {@link SocketExecuter} implementation this client will use.
+   * @param host The hostname or IP address to connect this client too.
+   * @param port The port to connect this client too.
+   * @throws IOException - This is thrown if there are any problems making the socket.
    */
-  public TCPClient(String host, int port, int timeout) {
-    maxConnectionTime = timeout;
-    this.host = host;
-    this.port = port;
+  public TCPClient(final SocketExecuter sei, final String host, final int port) throws IOException {
+    super(sei);
     remoteAddress = new InetSocketAddress(host, port);
+    channel = SocketChannel.open();
+    channel.configureBlocking(false);
   }
 
   /**
-   * <p>This creates a TCPClient based off an already existing {@link SocketChannel}.  This {@link SocketChannel} must already be 
-   * connected.</p>
+   * <p>This creates a TCPClient based off an already existing {@link SocketChannel}.  
+   * This {@link SocketChannel} must already be connected.</p>
    * 
+   * @param sei the {@link SocketExecuter} to use for this client.
    * @param channel the {@link SocketChannel} to use for this client.
    * @throws IOException if there is anything wrong with the {@link SocketChannel} this will be thrown.
    */
-  public TCPClient(SocketChannel channel) throws IOException {
-    maxConnectionTime = DEFAULT_SOCKET_TIMEOUT;
+  public TCPClient(final SocketExecuter sei, final SocketChannel channel) throws IOException {
+    super(sei);
     if(! channel.isOpen()) {
       throw new ClosedChannelException();
     }
     connectionFuture.setResult(true);
-    host = channel.socket().getInetAddress().getHostAddress();
-    port = channel.socket().getPort();
     if(channel.isBlocking()) {
       channel.configureBlocking(false);
     }
     this.channel = channel;
-    remoteAddress = channel.socket().getRemoteSocketAddress();
+    remoteAddress = (InetSocketAddress) channel.socket().getRemoteSocketAddress();
     startedConnection.set(true);
   }
   
-  /**
-   * This is used by {@link org.threadly.litesockets.tcp.SSLClient} to close the TCPClient object w/o closing its backing {@link SocketChannel}.
-   * This will make this {@link TCPClient} unusable.
-   */
-  protected void markClosed() {
-    if(closed.compareAndSet(false, true)) {
-      if(seb != null) {
-        seb.removeClient(this);
-      }
-    }
+  @Override
+  public void setConnectionTimeout(final int timeout) {
+    ArgumentVerifier.assertNotNegative(timeout, "Timeout");
```

Comment:
This should either be assertGreaterThanZero, or you could accept zero and in connect you could check that case and use Long.MAX_VALUE to disable a timeout.